tests/cases/compiler/narrowByEquality.ts

```diff
@@ -57,3 +58,13 @@ function test(level: number | string):number {
     }
     return 0;
 }
+
+// From issue #32798
+if (xUnknown == null) {
+    xUnknown;
+}
+
+if (xUnknown != null) {
+    xUnknown;
+}
```

Comment:
Could you also add an `else` branch here? Then we would be able to see that we are narrowing the type of `xUnknown` to be `null | undefined` when `xUnknown != null` is *false* (meaning `xUnknown == null` is true).
Those double negations are a bit tricky, I forgot to mention that test case in my last review.

Reply:
Ya that's totally fair! I decided to add it for both test cases just for my own sanity that it narrows it down correctly. Appreciate the help and guidance 😄